src/protonpack/java/com/codepoetics/protonpack/AggregatingSpliterator.java

```diff
@@ -15,8 +17,8 @@
     private List<I> currentSlide = new ArrayList<>();
 
     AggregatingSpliterator(Spliterator<I> source, BiPredicate<List<I>, I> predicate) {
-        this.source = source;
-        this.condition = predicate;
+        this.source = requireNonNull(source);
+        this.condition = requireNonNull(predicate);
```

Comment:
I like to follow the convention that constructors do no work (besides assignment), and cannot throw exceptions. So I would ordinary create a static factory method to do work like validation in.